app/views/buyers/applications/new.html.erb

```diff
@@ -32,6 +34,13 @@
 
 <script>
   document.addEventListener('DOMContentLoaded', function () {
-    window.planSelector()
+    // FIXME: get it already filtered from rails
+    var plans = JSON.parse(document.getElementById('plan_selector').getAttribute('plans'))
```

Comment:
Why not just output it as JSON here in the script? I'm not saying one is better, just interested.

If we keep doing this more and more, we should have a way to lazy load these data from our API, instead of populating it inside HTML.

Reply:
I'm still learning my way around ruby/rails. How would I do that?

Reply:
Which part? Serialize plans as json?

```js
var plans = #{json @plans};
```

Lazy loading from our API? Well, that is more complicated and should be more coordinated effort. But, basically something like that:

```js
var plans = fetch(#{json admin_api_application_plans_path(service_id: @service.id)})
```

Reply:
The serialized part, yes. But I thought that syntax was for slim..? In fact, it throws a JS error:
> Uncaught SyntaxError: Invalid or unexpected token

Reply:
Hm, my fault. JS != JSON and I haven't noticed this is erb.

That error message without the actual rendered html content is not much useful.

```js
var plans = JSON.parse(<%= json @plans %>);
```

Reply:
Finally:
```js
var plans = JSON.parse(<%= json @plans.to_json %>)
```